### PR TITLE
fix: restrict get_or_create to get_or_create_on_node

### PR DESCRIPTION
**Pull request by AloizioMacedo**
[LP:2147706](https://bugs.launchpad.net/maas/+bug/2147706) reports that `get_or_create` is erroring with multiple entries after commissioning an initial set of servers. The only way that this seems possible is if there are multiple interfaces with the same mac address, likely on different nodes since the issue happens after creating a first set of nodes and not on a clean environment.

The code in `update_physical_interface` does not account for this, but accounts for the possibility of the interface being in the incorrect node and does some cleanup related to this. This cleanup has duplicated logic in `get_or_create_on_node`, so we just use that instead (which should avoid the multiple entries issues due to the restriction on the node).

I took the liberty of also returning the information of whether the interface was created from `get_or_create_on_node`, to mirror what `get_or_create` does instead of having to do extra queries to check if something was added.

Resolves: [LP:2147706](https://bugs.launchpad.net/maas/+bug/2147706)

### CODE REVIEW

**Comment by maas-lander** [comment]
Check where you would like a Mattermost message to be sent to when CI completes and this PR is merged
- [ ] Direct message
- [ ] ~maas

**Comment by AloizioMacedo** [comment]
I put this in draft since I was not able to reproduce the issue yet. Will try a couple times, but the changes seem to be meaningful regardless.

File: src/maasserver/models/interface.py [hunk 1]
```diff
@@ -507,6 +507,7 @@ def get_or_create_on_node(
                 # current links because they are completely wrong.
                 interface.ip_addresses.all().delete()
                 interface.node_config = node.current_config
+            created = False
```
[the remark this answers is not in the document]
Comment (AloizioMacedo) [reply src/maasserver/models/interface.py:512]: True, but this is more of a bug in the original function regardless of this PR, rather than an issue of moving responsibilities now. For instance, this function is used elsewhere as well, and it is not saved from the caller side there.

Pushed the change.

File: src/metadataserver/builtin_scripts/network.py [hunk 2]
```diff
@@ -235,32 +235,19 @@ def update_physical_interface(
...
-                node.is_commissioning()
-                or (node.enable_hw_sync and port is not None)
-            ),
-        },
+
+    acquired = not (
+        node.is_commissioning() or (node.enable_hw_sync and port is not None)
+    )
+
+    interface, created = PhysicalInterface.objects.get_or_create_on_node(
```
Comment (tanzin8r) [inline src/metadataserver/builtin_scripts/network.py:243]: maybe add a regression test which sets up that duplicate state and runs the commissioning network update.
